ThomazFB/WordPress-Android
# GIF selection fully restored with Tenor API improvements

Summary
------------
This Pull Request brings integration for the Tenor API within the pre-existent GIF selection code inside the app. As described at [issue 11456](https://github.com/wordpress-mobile/WordPress-Android/issues/11456).

![screencapture-1585276984938 2020-03-26 23_46_37](https://user-images.githubusercontent.com/5920403/77717245-b89d3000-6fbe-11ea-8b8b-1b91699e1276.gif)

The solution proposes the following implementation:

* Test suite class added to `GiphyPickerDataSource` in order to better enforce the business rules when consuming any `GifProvider` implementation.

* The original integration with the `EditPostActivity` was restored, making GIF selection available again when writing or editing posts.

* Analytics tracking restored.

* The Giphy attribution icon was removed and replaced by the Tenor attribution icon.

Known missing points for upcoming Pull Requests
------------
* All Giphy packages and class naming is still missing, this will be the last modification to not storm the code review with name changes within the Pull Request
* All Giphy string translations still to be changed to Tenor

Testing
------------
1. Create a Tenor API key and paste it associated with the `wp.tenor.api_key` as the value inside your `gradle.properties`
2. Click to create a new post
3. Click on the overflow button at the top right end of the screen
4. select the `Switch to classic editor` 
5. Click on the plus (+) button at the bottom left end of the screen
6. Select the `Choose from Tenor` option
7. Type in the search bar to query GIFs
8. Once inside the Gif Picker, make sure that `selection`, `preview` and `add` are all working as expected


PR submission checklist
------------
